src/optimizer/join_order/cardinality_estimator.cpp

```diff
@@ -160,10 +195,137 @@ double CardinalityEstimator::EstimateCrossProduct(const JoinNode &left, const Jo
+			continue;
+		}
+
+		auto key = ColumnBinding(relation_id, column_bindings.at(it).column_index);
+		ColumnBinding value;
+		string column_name = "";
+		if (op_is_get) {
+			value = ColumnBinding(table_index, get->column_ids.at(it));
+			if (get->column_ids.at(it) == DConstants::INVALID_INDEX) {
+				column_name = "rowid";
+			} else {
+				column_name = get->names.at(get->column_ids.at(it));
+			}
+		} else {
+			bool relation_is_constant = false;
+			if (op_is_proj) {
+				// need to be careful here. A projection can project an amount of columns not present in an original
+				// get. here we can get the name properly, but to get the column, you need to go and get the column
+				// index of the logical get from where the column is originating from. loop through each expression and
+				// find the bindings. probably needd to cast it here.
+				switch (proj->expressions[it]->type) {
+				case ExpressionType::BOUND_COLUMN_REF: {
+					auto &column_ref = proj->expressions[it]->Cast<BoundColumnRefExpression>();
+					value = column_ref.binding;
+					break;
+				}
+				case ExpressionType::BOUND_FUNCTION: {
+					// look for a bound column ref in the function children. If you find one
+					// add the binding for that column ref for statistics.
+					auto &tmp = proj->expressions[it]->Cast<BoundFunctionExpression>();
+					auto found_child_column_ref = false;
+					for (auto &child : tmp.children) {
+						if (child->type == ExpressionType::BOUND_COLUMN_REF) {
+							auto &column_ref = child->Cast<BoundColumnRefExpression>();
+							value = column_ref.binding;
+							found_child_column_ref = true;
+							break;
+						}
+					}
+					if (!found_child_column_ref) {
+						relation_is_constant = true;
+					}
+					break;
+				}
+				default:
+					// the filter is not directly on a column or otherwise.
+					// set relation
+					// see ldbc-empty.test
+					relation_is_constant = true;
+				}
+
+				if (relation_is_constant) {
+					value = ColumnBinding(proj->table_index, it);
+					column_name = "Coalese Function";
+				} else {
+					auto get = GetDataRetOp(*proj, value);
+					if (get) {
+						auto &actual_get = get->Cast<LogicalGet>();
+						column_name =
+						    actual_get.names.at(actual_get.column_ids.at(value.column_index)) + " as " + proj->expressions[it]->GetName();
+
+						// sometimes a rowid is propagated as a column id, and affects the bindings.
+						// if we try to get statistics for a column when a row_id is present, our column id is off
+						// so we need to adjust our binding.
+						// see empty_joins.test
+						for (idx_t i = 0; i < actual_get.column_ids.size(); i++) {
+							if (actual_get.column_ids.at(i) == DConstants::INVALID_INDEX && i < value.column_index) {
+								value.column_index-=1;
+								break;
+							}
+						}
+					} else {
+						column_name = proj->expressions[it]->GetName();
+					}
+				}
+			}
+			// else could also be logical chunk get.
```

Comment:
I think these comments can be taken out. If it's a logical chunk get or some other non-reorder able join, we don't call this function